### DISCUSSION TITLE
RFC: drop the lambda_solid API

### DISCUSSION THREAD

**Discussion by mgasner** [dagster-io/dagster#3132] [Ideas]
The idea behind this API is that it smooths onboarding, but I think it's arguably more complicated to have it -- the front-end of the learning curve is very steep & adding any more concepts at the start isn't helping us.

**Comment by sryza** [comment]
I think this is a good idea.

**Comment by schrockn** [comment]
agreed. is there a good way to warn when someone uses a top-level import as opposed to dotting into something?

Sent via Superhuman ( https://sprh.mn/?vip=schrockn@elementl.com )

**Reply by mgasner** [reply]
i doubt it, i think we would need to hook into the import system which is quite invasive

**Reply by mgasner** [reply]
at a minimum i think we would need to edit `sys.meta_path` or equivalent in the top level `__init__.py` before importing anything ourselves... very ugly and potentially dangerous

**Comment by schrockn** [comment]
yeah maybe we can just put a blanket warning in the constructor warning that it will no longer be a top level import and be clear about how to disable.

Sent via Superhuman ( https://sprh.mn/?vip=schrockn@elementl.com )

**Reply by mgasner** [reply]
or we could have them be two different identical objects with the same name, and put the warning on one of them

**Reply by mgasner** [reply]
lol

**Comment by mgasner** [comment]
Hm, is the idea that we would also commit to maintain the dotted import? Or that this would become an internal API subject to change.
